pkg/oc/cli/admin/router/router.go

```diff
@@ -780,7 +781,13 @@ func RunCmdRouter(f kcmdutil.Factory, cmd *cobra.Command, out, errout io.Writer,
 		"STATS_PORT":                            strconv.Itoa(cfg.StatsPort),
 		"STATS_USERNAME":                        cfg.StatsUsername,
 		"STATS_PASSWORD":                        cfg.StatsPassword,
-		"ROUTER_THREADS":                        strconv.Itoa(int(cfg.Threads)),
+	}
+
+	// HAProxy defines the upper bound for nbthread as the number of bits in the C long type
+	if cfg.Threads > 0 && cfg.Threads <= uint32(unsafe.Sizeof(uint(0)))*8 {
+		env["ROUTER_THREADS"] = strconv.FormatUint(uint64(cfg.Threads), 10)
+	} else {
+		return fmt.Errorf("invalid value for threads must be between 0 and %d", unsafe.Sizeof(uint(0))*8)
```

Comment:
We don't want to throw this error when cfg.Threads=0 , you could do:
```
// Set the env var in the app.Environment
env := app.Environment{
...
  "ROUTER_THREADS":  strconv.FormatUint(uint64(cfg.Threads), 10),
}

if cfg.Threads > uint32(unsafe.Sizeof(uint(0)))*8  {
    return fmt.Errorf("invalid value for threads must be between 0 and %d", unsafe.Sizeof(uint(0))*8)
}
```

Reply:
What happens to haproxy process when nbthread > 64 (on 64 bit machine), will it complain and bail out?
Can you point to the haproxy docs or link that talks about the upper limit for nbthread variable (may be as a comment)?

Reply:
If nbthreads is bigger than sizeof(long) * 8 haproxy will complain, i didn't find this limit set on the documentation, here's the code:
https://github.com/haproxy/haproxy/blob/3e12304ae004284e3186b489f398908e59a8574d/include/common/hathreads.h#L134